### PR TITLE
Cvcreator sigmund

### PR DESCRIPTION
**Pull request by sigmunsl**
Now working with and without publications. Use flag -wp to get publications. Omits them by default.

### CODE REVIEW

File: cvcreator/templates/config.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+logo_width: 6cm
+logo_top: -2cm
+logo_left: 10cm
+logo_margin: 1cm
```
Comment (jonathf) [inline cvcreator/templates/config.yaml]: I am not sure if I agree that this should be added to git, but I don't mind either.

File: cvcreator/cvcreate.py [hunk 2]
```diff
@@ -36,7 +35,12 @@ def main():
     parser.add_argument(
         "a", metavar="a", type=int, nargs="*",
         help="Projects to include. Omit/0 for all/none.")
-
+    parser.add_argument(
+        "b", metavar="b", type=int, nargs="*",
+        help="Publications to include. Omit/0 for all/none")
+    parser.add_argument(
+        "-wp", "--with-publications", action="store_true", 
+        help="Publications to include.")
```
[the remark this answers is not in the document]
Comment (jonathf) [reply cvcreator/cvcreate.py]: Also, adding-all-on-omit for both projects and publications is the most sane behavior, as the two flags should behave in a similar manor.

I'll also make the claim that this practice is backwards compatible, as most of us do not have a publication list in our YAML files yet.

File: cvcreator/cvcreate.py [hunk 3]
```diff
@@ -95,7 +98,21 @@ def main():
...
+            
+            if 0 in args.b or not args.with_publications:
+                content.pop("Publications", None)
+            elif "Publications" not in content or not args.b:
+                pass
+            else:
+                pub = content.pop("Publications", {})
+                content["Publications"] = {}
+                i = 1
+                for n in args.b:
```
Comment (jonathf) [inline cvcreator/cvcreate.py]:
```suggestion
                for i, n in enumerate(args.b, start=1):
```

File: cvcreator/cvcreate.py [hunk 4]
```diff
@@ -95,7 +98,21 @@ def main():
...
+                    bn = "B%d" % n
+                    bi = "B%d" % i
+                    if an in pub:
+                        content["Publications"][ai] = proj.pop(an)
```
Comment (jonathf) [inline cvcreator/cvcreate.py]:
```suggestion
                        content["Publications"][ai] = proj.pop(an)
```

File: cvcreator/templates/config.yaml (hunk 1 above)
Comment (sigmunsl) [reply cvcreator/templates/config.yaml]: I agree. I'll remove it.

File: cvcreator/cvcreate.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (sigmunsl) [reply cvcreator/cvcreate.py]: I have implemented it now. Testing it for various cases to see if anything breaks. Thanks for the solid review!

File: cvcreator/cvcreate.py [hunk 5]
```diff
@@ -34,9 +39,11 @@ def main():
         "-s", '--silent', action="store_true",
         help="Muffle output.")
     parser.add_argument(
-        "a", metavar="a", type=int, nargs="*",
-        help="Projects to include. Omit/0 for all/none.")
-
+        "-p", "--projects", type=tuple_of_ints, 
+        default=(), help="Projects to include. Specify which entries by integers or use all to include all entries")
```
Comment (jonathf) [inline cvcreator/cvcreate.py]:
```suggestion
        default=(), help="Projects to include. Specify which entries by index or use 'all' to include all entries")
```

File: cvcreator/cvcreate.py [hunk 6]
```diff
@@ -34,9 +39,11 @@ def main():
...
+    parser.add_argument(
+        "-u", "--publications", type=tuple_of_ints, 
+        default=(), help="Publications to include. Specify which entris by integers or use all to inclue all entries.")
```
Comment (jonathf) [inline cvcreator/cvcreate.py]:
```suggestion
        default=(), help="Publications to include. Specify which entris by integers or use all to inclue all entries.")
```

File: cvcreator/txt2yaml.py [hunk 7]
```diff
@@ -152,6 +153,39 @@ def main():
                     a["Tools"] = line[6:]
                 else:
                     a[cur] += " " + line
+    
+        elif header[0] == "B":
+            
+            header = "B%d" % (int(header[1:-1])+1)
+            data["Publications"][header] = b = {}
+
+            if line[:8] == "Journal":
```
Comment (jonathf) [inline cvcreator/txt2yaml.py]: More idiomatic:
```suggestion
            entries = ["Journal", "Date", "Authors", "Title", "Summary", "DOI"]
            for entry in entries:
                if line.startswith(entry):
                    cur = entry
                    b[entry] = line[len(entry):]
                    break
            else:
                a[cur] += " " + line
```
Are you sure it should be `a` in the last line there?

**Pull request merged**